File: 2024/intermediate/05-palindrom/solution/solution.py
def jePalindrom(s):
    return s == s[::-1]
def najdiPalindrom(text):
    nejdelsi = ""
    for i in range(len(text)):
        for j in range(i + len(nejdelsi) + 1, len(text) + 1):
            if jePalindrom(text[i:j]) and len(text[i:j]) > len(nejdelsi):
                nejdelsi = text[i:j]

    return nejdelsi

print(najdiPalindrom(input()))

File: 2024/intermediate/05-palindrom/solution/solution.cs
using System;

class Program
{
    static bool JePalindrom(string s)
    {
        int length = s.Length;
        for (int i = 0; i < length / 2; i++)
        {
            if (s[i] != s[length - i - 1])
                return false;
        }
        return true;
    }

    static string NajdiPalindrom(string text)
    {
        string nejdelsi = "";
        for (int i = 0; i < text.Length; i++)
        {
            for (int j = i + nejdelsi.Length + 1; j <= text.Length; j++)
            {
                string sub = text.Substring(i, j - i);
                if (JePalindrom(sub) && sub.Length > nejdelsi.Length)
                {
                    nejdelsi = sub;
                }
            }
        }
        return nejdelsi;
    }

    static void Main()
    {
        Console.WriteLine(NajdiPalindrom(Console.ReadLine()));
    }
}